python/pyarrow/tests/test_pandas.py

```diff
@@ -1202,7 +1211,7 @@ def test_table_convert_date_as_object(self):
         df_datetime = table.to_pandas(date_as_object=False)
         df_object = table.to_pandas()
 
-        tm.assert_frame_equal(df.astype('datetime64[ns]'), df_datetime,
+        tm.assert_frame_equal(df.astype('datetime64[ms]'), df_datetime,
```

Comment:
Do we have coverage for testing that it stays nanoseconds if you specify `coerce_temporal_nanoseconds=True`?